README.md

```diff
@@ -1,81 +1,148 @@
+
+## Challenges for BPF in Kubernetes
+
+- Requires privileged pods
+  - BPF-enabled apps require at least CAP_BPF permissions and potentially more
+    depending on the type of program that is being attached.
+  - Since the Linux capabilities are very broad it is challenging to constrain a
+    pod to the minimum set of privileges required. This can allow them to do
+    damage (either unintentionally or intentionally).
+- Handling multiple BPF programs on the same BPF hooks
+  - Not all BPF hooks are designed to support multiple programs.
+  - Some software using BPF assumes exclusive use of a BPF hook and can
+    unintentionally eject existing programs when being attached.
+- Debugging problems with deployments is hard.
+  - The cluster administrator may not be aware that BPF programs are being used
+    in a cluster.
+  - It is possible for some BPF programs to interfere with others in
+    unpredictable ways.
+  - SSH access or a privileged pod is necessary to determine the state of BPF
+    programs on each node in the cluster.
+- There are no standard libraries for loading and managing BPF programs leaving
+  each application developer to implement their own.
+- Deployment on Kubernetes is not simple.
+	  - It is an involved process that requires first writing a daemon that loads
+	    your BPF bytecode and deploying it using a DaemonSet.
+	  - This requires careful design and intricate knowledge of the BPF program
+	    lifecycle to ensure your program stays loaded and that you can easily
+	    tolerate pod restarts and upgrades.
+
+## What is bpfd?
+
+bpfd is a software stack that aims to make it easy to load, unload, modify and
+monitor BPF programs whether on a single host, or in a Kubernetes cluster. bpfd comprises:
+
+  - bpfd: A system daemon that supports loading, unloading, modifying and
+	  monitoring of BPF programs exposed over a gRPC API.
+  - eBPF CRDS: That provide a way to express intent to load BPF programs
+	  `XdpProgram`, `TcProgram`, etc. and represent the runtime state of loaded
+	  programs via `BpfProgram`. 
+  - bpfd-agent: That ensures that the `BpfPrograms` for a given node are in the
+	  desired state.
+  - bpfd-operator: An operator, built using [Operator
+	  SDK](https://sdk.operatorframework.io/) that manages the installation and
+	  lifecycle of bpfd-agent and the CRDs in a Kubernetes cluster.
+
+bpfd is developed in Rust and built on top of Aya, a Rust BPF library.
+
+The benefits of this solution include the following:
+
+- Security
+  - Improved security because only the bpfd daemon, which can be tightly
+    controlled, has the privileges needed to load BPF programs, while access to
+    the API can be controlled via standard RBAC methods.  Within bpfd, only a
+    single thread keeps these capabilities while the other threads (serving
+    RPCs) do not.
+  - Gives the administrators control over who can load programs.
+  - Allows administrators to define rules for the ordering of networking BPF
```

Comment:
road map?

Reply:
Yes.  We plan to build policies around this, but it doesn't exist yet.